### PR TITLE
refactor: mpz-ot

### PR DESCRIPTION
**Pull request by sinui0**
This PR builds on #107 and #108 , refactoring our OT traits and implementations. It also _deletes_ the KOS actor now that we have the `mpz_common::sync::Mutex` primitive :pray: 

I haven't finished with the committed/verifiable functionalities on the shared KOS variants yet, but will punt that to another PR so we can just move on for now.

### CODE REVIEW

File: ot/mpz-ot/src/kos/shared_receiver.rs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+use utils_aio::non_blocking_backend::{Backend, NonBlockingBackend};
+
+use crate::{
+    kos::{Receiver, ReceiverError},
+    OTError, OTReceiver,
+};
+
+/// A shared KOS receiver.
+#[derive(Debug, Clone)]
+pub struct SharedReceiver<BaseOT> {
```
Comment (th4s) [inline ot/mpz-ot/src/kos/shared_receiver.rs:17]: Shouldn't these follow the new `Syncer` and `Mutex` approach?

**Review by th4s — APPROVED** [review]
gw :+1:

File: ot/mpz-ot/src/kos/shared_receiver.rs (hunk 1 above)
Comment (th4s) [reply ot/mpz-ot/src/kos/shared_receiver.rs:17]: I realized now that they do :sweat_smile: 

Can we eventually rename the `Mutex`. I am open to suggestions. What about `MpcMutex`?

Btw. this is an awesome abstraction :)

Comment (sinui0) [reply ot/mpz-ot/src/kos/shared_receiver.rs:17]: I like just calling it `Mutex`, prefixing it is a bit redundant. If it raises a reviewers eyebrow they just need to look that its imported from `mpz_common` and is documented :)

**Comment by sinui0** [comment]
Proceeding with merge as this should not introduce security issues, but @themighty1 if you get a time to sanity check where I've touched the coin-toss protocols in CO15 and KOS15 that would be appreciated :pray:

**Pull request merged**